### PR TITLE
Introduce assertions for StatusCode of HttpResponseMessage

### PR DESCRIPTION
**Pull request by mu88**
## IMPORTANT 

* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by a new or existing set of unit tests which follow the Arrange-Act-Assert syntax such as is used [in this example](https://github.com/fluentassertions/fluentassertions/blob/daaf35b9b59b622c96d0c034e8972a020b2bee55/Tests/FluentAssertions.Shared.Specs/BasicEquivalencySpecs.cs#L33).
* [x] If the contribution adds a feature or fixes a bug, please update [**the release notes**](https://github.com/fluentassertions/fluentassertions/tree/master/docs), which are published on the [website](https://fluentassertions.com/releases).
* [x] If the contribution changes the public API the changes needs to be included by running `AcceptApiChanges.ps1`/`AcceptApiChanges.sh`.
* [x] If the contribution affects [the documentation](https://github.com/fluentassertions/fluentassertions/tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).

### CODE REVIEW

File: Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+        [Theory]
+        [MemberData(nameof(GetClientErrorStatusCodes))]
+        public void Should_succeed_when_asserting_statuscode_is_client_error(HttpStatusCode statusCodeOfResponse)
+        {
+            // Arrange
+            var testee = new HttpResponseMessage(statusCodeOfResponse);
+
+            // Act / Assert
+            testee.Should().BeClientError();
```
Comment (dennisdoomen) [inline Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs:42]: 🤔 Considering that the error is a property of the response message, it feels to me that `HaveClientError` would be a more natural way of asserting this. Same with the other overloads that take an error.

File: Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+using System.Net.Http;
+using Xunit;
+using Xunit.Sdk;
+
+namespace FluentAssertions.Specs.Primitives
+{
+    public class HttpResponseMessageAssertionSpecs
+    {
+        [Theory]
+        [MemberData(nameof(GetSuccessStatusCodes))]
```
Comment (dennisdoomen) [inline Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs:13]: 🤔 I don't think it's very useful to run this test for every possible value. Just taking some sensible values, including edge cases would be much easier to understand. This also avoids the need to have a helper method, which I personally feel obfuscates the test purpose. Also applies to the other tests.

File: Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+                new HttpResponseMessage(HttpStatusCode.Gone).Should().BeSuccessful("because we want to test the failure {0}", "message");
+
+            // Assert
+            action
+                .Should().Throw<XunitException>()
+                .WithMessage("Expected HttpStatusCode to be successful (2xx) because we want to test the failure message, but found HttpStatusCode.Gone {value: 410}.");
+        }
+
+        [Fact]
+        public void Should_fail_when_asserting_statuscode_error_is_redirection()
```
Comment (dennisdoomen) [inline Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs:95]: 🔧 I suggest to group the success and failure cases for the same methods together.

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    {
+        public HttpResponseMessageAssertions(HttpResponseMessage value) => Subject = value;
+
+        /// <summary>
+        /// Gets the object which value is being asserted.
+        /// </summary>
+        public HttpResponseMessage Subject { get; }
+
+        /// <summary>
+        /// Asserts that the <see cref="HttpStatusCode"/> is successful.
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:36]: 🔧 I suggest to add the error number to the docs. Same for the other methods.

File: Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+using Xunit;
+using Xunit.Sdk;
+
+namespace FluentAssertions.Specs.Primitives
+{
+    public class HttpResponseMessageAssertionSpecs
+    {
+        [Theory]
+        [MemberData(nameof(GetSuccessStatusCodes))]
+        public void Should_succeed_when_asserting_statuscode_is_successful(HttpStatusCode statusCodeOfResponse)
```
Comment (dennisdoomen) [inline Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs:14]: 🙃 I'm nitpicking here, but `statuscode` consists of two separate words. I would also remove the `asserting` word to make the functional names more concise

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> BeSuccessful(string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .ForCondition(Subject.IsSuccessStatusCode)
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:48]: 🤔 What happens if `Subject` is `null`?

File: docs/_pages/httpresponsemessages.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+---
+title: HttpResponseMessages
```
Comment (dennisdoomen) [inline docs/_pages/httpresponsemessages.md:2]: ❌ This doesn't work, unless you also update https://github.com/fluentassertions/fluentassertions/blob/master/docs/_data/navigation.yml

**Review by dennisdoomen — CHANGES_REQUESTED** [review]
Thank you for the great contribution.

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs (hunk 6 above)
Comment (mu88) [reply Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:58]: What about introducing _Nullable Reference Types_?

**Comment by mu88** [comment]
Sorry about the build failure 🙄

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs (hunk 6 above)
Comment (dennisdoomen) [reply Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:58]: Unfortunately that's [much more complicated](https://github.com/fluentassertions/fluentassertions/issues/1115) than we though. For now, we've settled on specific errors.

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+    /// </summary>
+    [DebuggerNonUserCode]
+    public class HttpResponseMessageAssertions<TAssertions>
+        where TAssertions : HttpResponseMessageAssertions<TAssertions>
+    {
+        /// <summary>Initializes a new instance of the <see cref="HttpResponseMessageAssertions{TAssertions}"/> class.</summary>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="value"/> is <c>null</c></exception>
+        public HttpResponseMessageAssertions(HttpResponseMessage value)
+        {
+            Guard.ThrowIfArgumentIsNull(value, nameof(value));
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:36]: ❌ We still need to properly document the [design guidelines](https://github.com/fluentassertions/fluentassertions/issues/1340), but we use guards to capture misuse of our library. However, a `null` message is something that FA should be prepared for. In those cases, we use a proper `FailWith` construct, preferably per assertion method.

File: docs/_data/navigation.yml [hunk 9]
```diff
@@ -73,6 +73,8 @@ sidebar:
         url: /xml
       - title: Execution Time
         url: /executiontime
+      - title: HttpResponseMessage
```
Comment (dennisdoomen) [inline docs/_data/navigation.yml:76]: 🙃 I'd prefer something like `HTTP response messages`

File: docs/_pages/releases.md [hunk 10]
```diff
@@ -11,6 +11,7 @@ sidebar:
 
 ### What's New
 * Adding `ThatAreAsync()` and `ThatAreNotAsync()` for filtering in method assertions - [#1725](https://github.com/fluentassertions/fluentassertions/pull/1725)
+* Adding new assertions for the `HttpStatusCode` of a `HttpResponseMessage` - [#1698](https://github.com/fluentassertions/fluentassertions/discussions/1698)
```
Comment (dennisdoomen) [inline docs/_pages/releases.md:14]: ❌ Could you please point the link to the PR, not the issue.

**Review by dennisdoomen — CHANGES_REQUESTED** [review]
Almost there. Just some minor changes.

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+        /// </param>
+        public AndConstraint<TAssertions> BeRedirection(string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .ForCondition(Subject is not null)
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected HttpStatusCode to be redirection (3xx){reason}, but HttpResponseMessage was <null>.");
+
+            Execute.Assertion
+                .ForCondition((int)Subject.StatusCode >= 300 && (int)Subject.StatusCode <= 399)
```
Comment (jnyrup) [inline Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:78]: Suggestion
```cs
.ForCondition((int)Subject.StatusCode is >= 300 and <= 399)
```

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> HaveStatusCode(HttpStatusCode expected, string because = "", params object[] becauseArgs)
+        {
+            var success = Execute.Assertion
+                .ForCondition(Subject is not null)
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected HttpStatusCode to be {0}{reason}, but HttpResponseMessage was <null>.", expected);
+
+            if (success)
```
Comment (jnyrup) [inline Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:178]: This guard check should also be added to the other assertions.
It avoid an NRE when an assertion is wrapped in an `AssertionScope`, which defers failure until the disposal of the `AssertionScope`.

```cs
HttpResponseMessage subject = null;
using var _ = new AssertionScope();
subject.Should().BeSuccessful();

```

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+            : base(value)
+        {
+        }
+    }
+
+    /// <summary>
+    /// Contains a number of methods to assert that a <see cref="HttpResponseMessage"/> is in the expected state.
+    /// </summary>
+    [DebuggerNonUserCode]
+    public class HttpResponseMessageAssertions<TAssertions>
```
Comment (jnyrup) [inline Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:25]: This should probably either derive from `ReferenceTypeAssertions` or `ObjectAssertions`.
Deriving from `ObjectAssertions` will give the least invasive changes, as `HttpResponseMessage` currently should-resolves to `ObjectAssertions`.

File: Tests/Benchmarks/Benchmarks.csproj [hunk 14]
```diff
@@ -16,5 +16,6 @@
   </ItemGroup>
   <ItemGroup>
     <Reference Include="System.Data.DataSetExtensions" Condition="'$(TargetFramework)' == 'net472'" />
+    <Reference Include="System.Net.Http" Condition="'$(TargetFramework)' == 'net472'" />
```
Comment (jnyrup) [inline Tests/Benchmarks/Benchmarks.csproj:19]: Why is this needed?

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs (hunk 11 above)
Comment (dennisdoomen) [reply Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:78]: I love C# 9.0 ;-)

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> BeSuccessful(string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:47]: ❌ If the call to `BeSuccessful` was encapsulated by an `AssertionScope`, the `FailWith` will just queue an error message on the current scope, but not throw an actual exception. You either need to chain the two assertions using `Then` or account for the boolean return value of `FailWith`. Same for the other tests.

**Review by dennisdoomen — CHANGES_REQUESTED** [review]

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs (hunk 11 above)
Comment (mu88) [reply Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs]: Yes me too! They just have to be integrated into my muscle memory ;)

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs (hunk 12 above)
Comment (mu88) [reply Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:190]: Thanks for the explanation!

File: Tests/Benchmarks/Benchmarks.csproj (hunk 14 above)
Comment (mu88) [reply Tests/Benchmarks/Benchmarks.csproj:19]: With the "new" .NET (.NET Standard, .NET Core, .NET ≥ 5.0), `System.Net.Http` has become part of .NET's _Base Class Libraries_. Therefore the assembly is automatically and implicitly referenced in all newer projectes.

But with .NET Framework, this is not the case. The assembly is always present in the GAC (because the .NET Framework installs it), but it has to be referenced manually.

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs [hunk 16]
```diff
@@ -0,0 +1,234 @@
...
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> HaveServerError(string because = "", params object[] becauseArgs)
+        {
+            var success = Execute.Assertion
+                .ForCondition(Subject is not null)
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected HttpStatusCode to be server error (5xx){reason}, but HttpResponseMessage was <null>.");
+
+            if (success)
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:161]: 🔧 You can also do this by chaining those calls together. See https://github.com/fluentassertions/fluentassertions/blob/42f0f8f001cc8f291ae6d3c6efac1bf0da34100f/Src/FluentAssertions/Collections/GenericCollectionAssertions.cs#L115

**Review by dennisdoomen — APPROVED** [review]

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs (hunk 16 above)
Comment (mu88) [reply Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:161]: Would you prefer that pattern?

Comment (dennisdoomen) [reply Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:161]: With a 🔧 I mean it's a suggestion. Up to you if you still want to do it. I've already approved the PR

**Comment by mu88** [comment]
Do you need anything from my side to close this PR, @dennisdoomen ? Or will you resolve the conflicts when creating the next release?

**Comment by dennisdoomen** [comment]
I'm waiting for the approval of @jnyrup

**Comment by jnyrup** [comment]
Sorry for the delay.
I wanted to fiddle a bit with this to fully understand why a project reference to `System.Net.Http` is needed as I would hate to force an extra dependency on consumers of this library.
I should have time this week to test.

**Comment by dennisdoomen** [comment]
And I thought you were supposed to be off your computer this week ;-)

> Yuck!

It's [really](https://github.com/Microsoft/dotnet/blob/master/releases/net472/KnownIssues/613745%20-%20Single-name%20references%20are%20removed%20by%20the%20SDK%20when%20targeting%204.7.2.md) ugly... 

> If we want to make sure that one can upgrade without having to manually adding a reference to `System.Net.Http` we could conditionally add a `PackageReference` for net47 and cross our fingers that it doesn't break anything else for consumers.

I've tried creating a .NET 4.7.0, .NET 4.7.2 and a .NET 6 project. For the first two, I had to add an assembly reference to refer to an `HttpResponseMessage`. For the latter, it was just there. 

I'm fine to take the plunge and add that conditional package reference.

**Comment by dennisdoomen** [comment]
@jnyrup do you stick with your opinion?

**Comment by jnyrup** [comment]
Let's conditionally add `System.Net.Http` as a nuget dependency to the net47 TFM.
```cs
<PackageReference Include="System.Net.Http" Version="4.3.4" />
```

As the last [link](https://github.com/Microsoft/dotnet/blob/master/releases/net472/KnownIssues/613745%20-%20Single-name%20references%20are%20removed%20by%20the%20SDK%20when%20targeting%204.7.2.md) mentions there's a workaround and even though the issue doesn't seem to have been updated accordingly a newer version of MSBuild should have a proper fix.

**Comment by dennisdoomen** [comment]
@mu88 can you move this further?

**Comment by mu88** [comment]
@dennisdoomen Sure! I'll do this next week after the holidays

**Comment by mu88** [comment]
@dennisdoomen & @jnyrup : good for you this way?

File: docs/_pages/releases.md [hunk 17]
```diff
@@ -4,15 +4,27 @@ permalink: /releases/
...
 classes: wide
 sidebar:
-  nav: "sidebar"
+nav: "sidebar"
 ---
 
 ## Unreleased
 
+### What's New
+
```
Comment (dennisdoomen) [inline docs/_pages/releases.md:13]: 🤔 Did you forget to add the release notes?

**Review by dennisdoomen — APPROVED** [review]

File: docs/_pages/releases.md [hunk 18]
```diff
@@ -4,15 +4,27 @@ permalink: /releases/
 layout: single
 classes: wide
 sidebar:
-  nav: "sidebar"
+nav: "sidebar"
```
Comment (jnyrup) [inline docs/_pages/releases.md:7]: Was this change intentional?
It seems to change the generated html

File: Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs [hunk 19]
```diff
@@ -0,0 +1,442 @@
...
+        {
+            // Act
+            Action action = () =>
+                new HttpResponseMessage(HttpStatusCode.Gone).Should().BeSuccessful("because we want to test the failure {0}", "message");
+
+            // Assert
+            action
+                .Should().Throw<XunitException>()
+                .WithMessage("Expected HttpStatusCode to be successful (2xx) because we want to test the failure message, but found HttpStatusCode.Gone {value: 410}.");
+        }
```
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs:47]: I know we're not consistent about this in the tests, but let's combine these two tests to a single one.
Verifying that `because` and `becauseArgs` are included in the failure message is part of the behavior when failing.

File: docs/_pages/httpresponsemessages.md [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+permalink: /httpresponsemessages/
+layout: single
+classes: wide
+sidebar:
+nav: "sidebar"
```
Comment (jnyrup) [inline docs/_pages/httpresponsemessages.md:7]: If this should be consistent with other markdown files.
```suggestion
  nav: "sidebar"
```

**Review by jnyrup — COMMENTED** [review]
Mostly informational, but I stumbled upon this the other day.

https://github.com/balanikas/FluentAssertions.Http
https://www.nuget.org/packages/fluentassertions.http

File: docs/_pages/releases.md (hunk 18 above)
Comment (mu88) [reply docs/_pages/releases.md:7]: Nope, stupid mistake from my side, sorry about that

File: docs/_pages/httpresponsemessages.md (hunk 20 above)
Comment (mu88) [reply docs/_pages/httpresponsemessages.md:7]: Was related to the other mistake

File: Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs (hunk 19 above)
Comment (mu88) [reply Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs:47]: But combining only these two tests makes it imho less consistent 🙂 wouldn't it be better to combine the other tests as well?

Comment (jnyrup) [reply Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs:47]: Bad phrasing from my side.
Yes, let's combine each pair of `Throw<XunitException>()` and `Throw<XunitException>().WithMessage()` tests.

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs [hunk 21]
```diff
@@ -0,0 +1,234 @@
...
+                    .ForCondition(IsServerError())
+                    .BecauseOf(because, becauseArgs)
+                    .FailWith("Expected HttpStatusCode to be server error (5xx){reason}, but found {0}.", Subject.StatusCode);
+            }
+
+            return new AndConstraint<TAssertions>((TAssertions)this);
+        }
+
+        /// <summary>
+        /// Asserts that the value is equal to the specified <paramref name="expected"/> value.
```
Comment (jnyrup) [inline Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:173]:
```suggestion
        /// Asserts that the <see cref="HttpStatusCode"/> is equal to the specified <paramref name="expected"/> value.
```

File: Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs [hunk 22]
```diff
@@ -0,0 +1,234 @@
...
+                    .ForCondition(Subject!.StatusCode == expected)
+                    .BecauseOf(because, becauseArgs)
+                    .FailWith("Expected HttpStatusCode to be {0}{reason}, but found {1}.", expected, Subject.StatusCode);
+            }
+
+            return new AndConstraint<TAssertions>((TAssertions)this);
+        }
+
+        /// <summary>
+        /// Asserts that the value is not equal to the specified <paramref name="unexpected"/> value.
```
Comment (jnyrup) [inline Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs:202]:
```suggestion
        /// Asserts that the <see cref="HttpStatusCode"/> is not equal to the specified <paramref name="unexpected"/> value.
```

File: docs/_pages/releases.md [hunk 23]
```diff
@@ -11,6 +11,7 @@ sidebar:
 
 ### What's New
 * Adding `ThatAreAsync()` and `ThatAreNotAsync()` for filtering in method assertions - [#1725](https://github.com/fluentassertions/fluentassertions/pull/1725)
+* Adding new assertions for the `HttpStatusCode` of a `HttpResponseMessage` - [#1737](https://github.com/fluentassertions/fluentassertions/pull/1737)
```
Comment (jnyrup) [inline docs/_pages/releases.md:14]:
```suggestion
* Adding new assertions for the `HttpStatusCode` of an `HttpResponseMessage` - [#1737](https://github.com/fluentassertions/fluentassertions/pull/1737)
```

**Review by jnyrup — COMMENTED** [review]
Just a few cosmetic changes before smashing the merge button.

**Review by jnyrup — APPROVED** [review]

**Comment by dennisdoomen** [comment]
Please rebase the PR on the latest `master`.

File: docs/_pages/releases.md [hunk 24]
```diff
@@ -11,6 +11,7 @@ sidebar:
 
 ### What's New
 * Adding `ThatAreStatic()` and `ThatAreNotStatic()` for filtering in method assertions - [#1740](https://github.com/fluentassertions/fluentassertions/pull/1740)
+* Adding new assertions for the `HttpStatusCode` of an `HttpResponseMessage` - [#1737](https://github.com/fluentassertions/fluentassertions/pull/1737)
```
Comment (lg2de) [inline docs/_pages/releases.md:14]: How about adding a hint for the new reference (in net47) to `System.Net.Http`?

Comment (dennisdoomen) [reply docs/_pages/releases.md:14]: You mean in the release notes?

Comment (lg2de) [reply docs/_pages/releases.md:14]: Yes. For 6.X.Y, there will not be any migration guide, isn't it?

Comment (dennisdoomen) [reply docs/_pages/releases.md:14]: No, because there's nothing to migrate.

Comment (lg2de) [reply docs/_pages/releases.md:14]: Then this would be the only place to inform the user.
This is not a breaking change but in an environment of restricted nuget access (like my company's package filter) it would be good to know before.

Comment (dennisdoomen) [reply docs/_pages/releases.md:14]: Hmm, that's kind of edgy case. Are you saying you have to get approval to take a dependency on Microsoft packages? If so, feel free to propose a PR that captures this.
